lore/developers/contributing/designs/network-design.md

```diff
@@ -0,0 +1,76 @@
+---
+name: network design
+layout: default
+---
+
+# BonnyCI Cloud Network Design
+
+## Problem Statement
+
+Before we redeploy BonnyCI to our new cloud environment, we need to rethink our network design choices.  On our existing cloud, we are running all services and hosts on a single flat internal network.  This allows all hosts to communicate with one another, which is less than ideal considering some hosts (the nodepool slaves) will be executing arbitrary code from third party repositories and patches.  We need to design our network topology in a way that provides the bare minimum required connectivity both internally between hosts and from the outside world in.
+
+## Overview
+
+We will utilize two separate internal networks to isolate our production services from slave nodes that will be executing tests.  The two networks will be connected by a L3 router which will provide connectivity between internal networks as well outbound connectivity to the internet.  Security groups will be used to limit connectivity from the outside world, allow connectivity between hosts on the same network, and prohibit connectivity from slave nodes to the rest of the infrastructure.  This network topology will be configured in Neutron via Ansible automation and will be scoped to production project.  It should be trivial for developers to recreate this topology within their own tenant and project.
+
+### Projects
+
+Network resources will be created within a ``control-plane`` project.  We will be utilizing shared networks, which must be created by a user with the ``cloud_admin`` role on that project. Compute resources for control plane services will be created within the ``control-plane`` project.  Nodepool itself will be configured to use a specific ``nodepool`` project for creating its instances with interfaces attached to a shared network that was created on the ``control-plane`` project.
+
+### Networks
+
+We will create two separate internal networks and hosts will be given interfaces on them depending on the role of said host.
+
+ * ``control-plane 192.168.10.0/24`` - This network will be used for communication between all of our production services, as well as automation between the bastion host and our producion servers.  As per our current infrastructure, the following static hosts will have interfaces on this network and internal DNS updated accordingly:
```

Comment:
Per our [Markdown lint configuration](https://github.com/BonnyCI/bonnyci.org/blob/master/.markdownlint.js), unordered lists should start with an asterix, and nested unordered lists are indented by two spaces.

I would suggest rewriting this sections as:

```markdown
* `control-plane 192.168.10.0/24` - This network will be used for communication between all of our production services, as well as automation between the bastion host and our producion servers.  As per our current infrastructure, the following static hosts will have interfaces on this network and internal DNS updated accordingly:
  * bastion.bonncyi-internal.portbleu.com
  * logs.bonnyci-internal.portbleu.com
  * merger01.bonnyci-internal.portbleu.com
  * nodepool.bonncyi-internal.portbleu.com
  * zuul.bonnyci-internal.portbleu.com
* `nodepool 10.0.0.0/8` - This is an shared network that will be used by nodepool slaves. None of the static service hosts within the control plane will receive an interface on this network.
```

Which would be rendered as:

* `control-plane 192.168.10.0/24` - This network will be used for communication between all of our production services, as well as automation between the bastion host and our producion servers.  As per our current infrastructure, the following static hosts will have interfaces on this network and internal DNS updated accordingly:
  * bastion.bonncyi-internal.portbleu.com
  * logs.bonnyci-internal.portbleu.com
  * merger01.bonnyci-internal.portbleu.com
  * nodepool.bonncyi-internal.portbleu.com
  * zuul.bonnyci-internal.portbleu.com
* `nodepool 10.0.0.0/8` - This is an shared network that will be used by nodepool slaves. None of the static service hosts within the control plane will receive an interface on this network.